internal/utils/message_hashmap.go

```diff
@@ -24,15 +23,11 @@ func NewMessageHashesMap() *MessageHashesMap {
 // Insert inserts the hash of the current message literal into the map and return true if an existing value was already
 // present.
 func (m *MessageHashesMap) Insert(id imap.InternalMessageID, literal []byte) (bool, error) {
-	hash := sha256.New()
-
-	if _, err := hash.Write(literal); err != nil {
+	literalHashStr, err := rfc822.GetMessageHash(literal)
+	if err != nil {
```

Comment:
if `err!=nil` then try `literalHash` as before?

Reply:
We need to parse here again sadly, since the result of the previous parsing is lost.